feat: track bitfi fees & revenue

starfrich opened the pull request:
#4793

llamabutler commented:
The bitfi.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees bitfi.ts
    
    🦙 Running BITFI.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 03 Dec 2025 09:43:05 GMT
    End Date:	Thu, 04 Dec 2025 09:43:05 GMT
    ---------------------------------------------------
    
    chain     | Daily fees | Daily user fees | Daily revenue | Daily protocol revenue | Start Time
    ---       | ---        | ---             | ---           | ---                    | ---       
    ethereum  | 0.00       | 0.00            | 0.00          | 0.00                   | 6/2/2025  
    bsc       | 936.00     | 936.00          | 936.00        | 936.00                 | 6/2/2025  
    base      | 0.00       | 0.00            | 0.00          | 0.00                   | 6/2/2025  
    btr       | 0.00       | 0.00            | 0.00          | 0.00                   | 6/2/2025  
    hemi      | 45.00      | 45.00           | 45.00         | 45.00                  | 6/2/2025  
    core      | 0.00       | 0.00            | 0.00          | 0.00                   | 6/2/2025  
    goat      | 0.00       | 0.00            | 0.00          | 0.00                   | 6/2/2025  
    Aggregate | 981.00     | 981.00          | 981.00        | 981.00                 |

llamabutler commented:
The bitfi-btc.ts adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees bitfi-btc.ts
    
    🦙 Running BITFI-BTC.TS adapter 🦙
    ---------------------------------------------------
    Start Date:	Fri, 05 Dec 2025 11:46:44 GMT
    End Date:	Sat, 06 Dec 2025 11:46:44 GMT
    ---------------------------------------------------
    
    chain     | Daily fees | Daily user fees | Daily revenue | Daily protocol revenue | Start Time
    ---       | ---        | ---             | ---           | ---                    | ---       
    ethereum  | 0.00       | 0.00            | 0.00          | 0.00                   | 17/9/2024 
    bsc       | 0.00       | 0.00            | 0.00          | 0.00                   | 26/12/2024
    base      | 0.00       | 0.00            | 0.00          | 0.00                   | 21/5/2025 
    btr       | 0.00       | 0.00            | 0.00          | 0.00                   | 3/1/2025  
    hemi      | 0.00       | 0.00            | 0.00          | 0.00                   | 14/3/2025 
    core      | 0.00       | 0.00            | 0.00          | 0.00                   | 18/6/2025 
    goat      | 0.00       | 0.00            | 0.00          | 0.00                   | 24/6/2025 
    Aggregate | 0.00       | 0.00            | 0.00          | 0.00                   |

starfrich commented:
> @starfrich is this listing for https://defillama.com/protocol/bitfi-btc or something else?
> 
> i can see you have added start date as 2025-02-06, is there anyway to refill for full history for older version? lmk if you need any help

1. yes, this is for https://defillama.com/protocol/bitfi-btc. i've renamed the file from `bitfi.ts` to `bitfi-btc.ts` to match the protocol slug.

2. i've corrected the start dates, they were incorrectly set to 2025-02-06 for all chains. historical data can be backfilled from the actual contract deployment dates:
     - ethereum bfBTC: 2024-09-17
     - ethereum bfUSD: 2025-11-03
     - other chains: verified from their respective block explorers
     
the adapter uses event logs which are permanently stored on-chain, so all fee data since deployment can be indexed.

treeoflife2 commented:
@starfrich can you create new listing to track the bfUSD fees too(https://defillama.com/protocol/bitfi-basis)? as i can see you already have it in the adapter file

starfrich commented:
checking in here, happy to iterate on this if there’s any feedback. thanks
